qiskit_nature/results/utils/protein_decoder.py

```diff
@@ -0,0 +1,109 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An auxiliary class that gets the turns in the main and side chain of a molecule
+ in ProteinFoldingResult."""
+from typing import List, Tuple, Optional
+
+
+class ProteinDecoder:
+
+    """This class handles the decoding of the compact solution in ProteinFoldingProblem
+    and returns the information encoded in the result about the turns
+    associated to the main and side chains.
+    """
+
+    def __init__(
+        self, best_sequence: str, side_chain_hot_vector: List[bool], fifth_bit: bool
+    ) -> None:
+        """
+        Args:
+            best_sequence: Will be the sequence decoded.
+            side_chain_hot_vector: boolean list with the position of the side chains
+            fifth_bit: True if the fifth bit has defaulted to 1.
```

Comment:
I would copy more explanation what the `fifth_bit` is and what role it plays. It is probably documented somewhere in the PF problem code.